pybwa/libbwaaln.pyx

```diff
@@ -64,56 +64,65 @@ cdef class BwaAlnOptions:
+    # https://github.com/lh3/bwa/blob/b92993c1161e73167181558856567ef2f367e3f0/bwtaln.c#L24
+    #
+    # Mappings between the properties of this class and the members of the `gap_opt_t` struct are as
+    # follows:
+    #
+    # `BwaAlnOptions` property                    | `gap_opt_t` member
+    # ----------------------------------------------------------------
+    # max_mismatches                              | max_diff
+    # max_gap_opens                               | max_gapo
+    # max_gap_extensions                          | max_gape
+    # min_indel_to_end_distance                   | indel_end_skip
+    # max_occurrences_for_extending_long_deletion | max_del_occ
+    # seed_length                                 | seed_len
+    # max_mismatches_in_seed                      | max_seed_diff
+    # mismatch_penalty                            | s_mm
+    # gap_open_penalty                            | s_gapo
+    # gap_extension_penalty                       | s_gape
+    # stop_at_max_best_hits                       | max_top2
+    # max_hits                                    | <not a member>
+    # log_scaled_gap_penalty                      | mode (bitwise flag)
+    # find_all_hits                               | mode (bitwise flag)
+    # with_md                                     | <not a member>
+    # threads                                     | n_threads
+
     def __init__(self,
-                 max_mismatches: int | None = None,
-                 max_gap_opens: int | None = None,
-                 max_gap_extensions: int | None = None,
-                 min_indel_to_end_distance: int | None = None,
-                 max_occurrences_for_extending_long_deletion: int | None = None,
-                 seed_length: int | None = None,
-                 max_mismatches_in_seed: int | None = None,
-                 mismatch_penalty: int | None = None,
-                 gap_open_penalty: int | None = None,
-                 gap_extension_penalty: int | None = None,
-                 stop_at_max_best_hits: int | None = None,
-                 max_hits: int | None = 3,
-                 log_scaled_gap_penalty: bool | None = None,
-                 find_all_hits: bool | None = None,
-                 with_md: bool | None = False,
-                 threads: int | None = None
+                 max_mismatches: int = -1,
+                 max_gap_opens: int = 1,
+                 max_gap_extensions: int = 6,
+                 min_indel_to_end_distance: int = 5,
+                 max_occurrences_for_extending_long_deletion: int = 10,
+                 seed_length: int = 32,
+                 max_mismatches_in_seed: int = 2,
+                 mismatch_penalty: int = 3,
+                 gap_open_penalty: int = 11,
+                 gap_extension_penalty: int = 4,
+                 stop_at_max_best_hits: int = 30,
+                 max_hits: int | None = None,
```

Comment:
@nh13 I couldn't find where `max_hits` was being initialized previously, and it doesn't have an equivalent member in `gap_opt_t`. What should we set as the default here?